### ISSUE TITLE
html page builder toggle

### ISSUE THREAD

**Issue by woutersf** [neocities/neocities#452]
first: I love the product! 
And I would like my kids to get involved in this too (8y/10y). They don't know html however. 
As a kid myself I learned html trough microsoft frontpage, let's not add that to the stack. 

I would love to suggest this feature: a nicer page builder. 
It could be toggled with a simple button like on this position: 
<img width="957" alt="Screenshot 2023-06-02 at 17 39 50" src="https://github.com/neocities/neocities/assets/592312/031bfcb1-4495-432d-a7a2-b9392c0723dc">
And then something could open like grapesJS page builder ([DEMO](https://grapesjs.com/demo.html))
![image](https://github.com/neocities/neocities/assets/592312/8be1ae8d-79ac-47ec-b2a1-8748b2efc845)

I will try to commit this myself. But I'm a neocities noob so this might take some time.

**Comment by xandracc** [comment]
please no. i learned HTML when i was 8 years old and was building websites using notepad. they're entirely capable of learning if they want to.

if you want an editor like this for your kids, have them play around in wix or weebly or wordpress with a theme and the elementor plugin. please don't turn neocities into this.

**Comment by woutersf** [comment]
Thanks for your response. 
This reaction surprised me a little bit. 

I don't understand the _please no._ 
Please no because you don't want kids on the site? 
Please no because you don't want them to play with html in a sort of editor here?
What's the difference with them using another tool and then copy pasting the HTML in? 

I also don't understand what you mean by 
_please don't turn neocities into this._
Could you elaborate what you mean by this?

**Comment by xandracc** [comment]
no, kids _should_ learn how to code. the social aspect would be something for parents to watch, but i'm referring to your WYSIWYG editor. instead of turning neocities into the WYSIWYG pagebuilder you're referring to, kids can play on wix or weebly or another editor that's out there that utilizes a WYSIWYG webpage editor.

there's no difference between doing it in something else and bringing that into neocities, but neocities currently does a great job of exposing people to learning HTML by doing and providing a social environment that helps users with inspiration, connection, and skill-building. WYSIWYG editors available for free on neocities will end up flooding the site with commercial sites due to its free hosting, including dubious and fraudulent commercial endeavors (e.g. fraudulent dropshipping). right now, the barrier is high enough that folks who are doing those types of websites are using template editors in other site-building services (wix, weebly) and not putting in the effort of coding things themselves, so the amount of commercial content on neocities is very small in comparison to the personal websites that currently overwhelmingly call the hosting service home.

the social aspect makes neocities different than other web-hosting providers. and i think what is currently on the market (wix , squarespace, weebly, etc.) is better for where those services should be such as WYSIWYG editors, which can then be utilized to play in for kids, rather than try to put a new feature that is antithetical to the overall message of the site on a service that is better suited for a missing part of the market, in this case: free web-hosting that allows code-first rather than template-first design.

hope this better explains my position. let me know if i can clarify anything.

**Comment by debil03311** [comment]
> What's the difference with them using another tool and then copy pasting the HTML in?

None, but you shouldn't be doing that on Neocities in the first place. It doesn't advertise itself as that sort of platform because it isn't. It's a platform that allows and encourages you to develop your own static website from the ground up, and to draw inspiration from other like-minded users.

You can learn many things about development by mashing together different bits and bobs of code from different places, but a WYSIWYG editor barely exposes you to the underlying code, if at all, and more often than not generates its own terribly formatted "soulless" code which no one wants to even look at.

Imagine the following scenario: you see something you like on a website, and would like to implement it on your own. You open the devtools or source code view, and are greeted by senseless, auto-generated and undocumented markup vomit. What can you learn from that?

> I also don't understand what you mean by _please don't turn neocities into this._ Could you elaborate what you mean by this?

This is just not what Neocities is for, you're given a barebones setup for a reason. In a sea of codeless drag-and-drop website builders, Neocities feels like a breath of fresh air for those interested in actual hands-on web development.

**Comment by kyledrake** [comment]
Yeah we're never going to get into the WYSIWYG editor space, there's just no good way to mix it with handcoding HTML
